modules/logging/filters.go

```diff
@@ -185,15 +187,117 @@ func (m IPMaskFilter) Filter(in zapcore.Field) zapcore.Field {
 	return in
 }
 
+type ActionType string
+
+const (
+	// Replace value(s) of query parameter(s).
+	ReplaceType ActionType = "replace"
+	// Delete query parameter(s).
+	DeleteType ActionType = "delete"
+)
+
+type queryFilterAction struct {
+	// `replace` to replace the value(s) associated with the parameter(s) or `delete` to remove them entirely.
+	Type ActionType `json:"type"`
+	// The name of the query parameter.
+	Parameter string `json:"parameter"`
+	// The value to use as replacement if the action is `replace`.
+	Value string `json:"value,omitempty"`
+}
+
+// QueryFilter is a Caddy log field filter that
+// filters query parameters from an URL.
+// It updates the filtered URL to remove or replace query parameters containing sensitive data.
+// For instance, it can be used to redact OAuth access tokens, session IDs, magic links tokens
+// or JWT when passed as query parameters (which is generally considered a bad practice, but cannot
+// always be avoided).
```

Comment:
Since this will get rendered as Markdown, having a newline between will actually turn them into two paragraphs.

I shortened the line length a bit too, so the comment has a nicer shape 😅 and a few minor changes to wording. I removed the JWT bit because it's a bit of an unnecessary "aside".

```suggestion
// QueryFilter is a Caddy log field filter that filters
// query parameters from a URL.
//
// This filter updates the logged URL string to remove or replace query
// parameters containing sensitive data. For instance, it can be used
// to redact any kind of secrets which were passed as query parameters,
// such as OAuth access tokens, session IDs, magic link tokens, etc.
```